src/agent/src/rpc.rs

```diff
@@ -1432,23 +1472,39 @@ fn update_container_namespaces(
     // Else set this to empty string so that a new pid namespace is
     // created for the container.
     if sandbox_pidns && sandbox.sandbox_pidns.is_some() {
-        pid_ns.path = String::from(sandbox.sandbox_pidns.as_ref().unwrap().path.as_str());
+        pid_ns.path = String::from(
+            sandbox
+                .sandbox_pidns
+                .as_ref()
+                .ok_or("failed to get sandbox ref")
+                .map_err(|e| anyhow!(e))?
+                .path
+                .as_str(),
+        );
     }
 
     linux.namespaces.push(pid_ns);
     Ok(())
 }
 
-fn append_guest_hooks(s: &Sandbox, oci: &mut Spec) {
+fn append_guest_hooks(s: &Sandbox, oci: &mut Spec) -> Result<()> {
     if s.hooks.is_none() {
-        return;
+        return Ok(());
     }
-    let guest_hooks = s.hooks.as_ref().unwrap();
+
+    let guest_hooks = s
+        .hooks
+        .as_ref()
+        .ok_or("failed to get guest hook reference")
+        .map_err(|e| anyhow!(e))?;
+
```

Comment:
Again the `is_none()` case is already dealt with so you want an `if let` (or `match`).